snapcraft_legacy/plugins/v2/kernel.py

```diff
@@ -0,0 +1,1766 @@
+      in the install directory of the current part. This flag allows used of firmware
+      from stage directory instead.
+
+    - kernel-initrd-firmware:
+      (array of string; default: none)
+      list of firmware files to be included in the initrd; these need to be
+      relative paths to stage directory.
+      <stage/part install dir>/firmware/* -> initrd:/lib/firmware/*
+
+    - kernel-initrd-compression:
+      (string; default: as defined in ubuntu-core-initrd(lz4)
+      initrd compression to use; the only supported values now are
+      'lz4', 'xz', 'gz', 'zstd'.
+
+    - kernel-initrd-compression-options:
+      Optional list of parameters to be passed to compressor used for initrd
+      (array of string): defaults are
+        gz:  -7
+        lz4: -9 -l
+        xz:  -7
+        zstd: -1 -T0
+
+    - kernel-initrd-overlay
+      (string; default: none)
+      Optional overlay to be applied to built initrd.
+      This option is designed to provide easy way to apply initrd overlay for
+      cases modifies initrd scripts for pre uc20 initrds.
+      Value is relative path, in stage directory. and related part needs to be
+      built before initrd part. During build it will be expanded to
+      ${SNAPCRAFT_STAGE}/{initrd-overlay}
+
+    - kernel-initrd-addons
+      (array of string; default: none)
+      Optional list of files to be added to the initrd.
+      Function is similar to kernel-initrd-overlay, only it works on per file
+      selection without a need to have overlay in dedicated directory.
+      This option is designed to provide easy way to add additional content
+      to initrd for cases like full disk encryption support, when device
+      specific hook needs to be added to the initrd.
+      Values are relative path from stage directory, so related part(s)
+      need to be built before kernel part.
+      During build it will be expanded to
+      ${SNAPCRAFT_STAGE}/{initrd-addon}
+      Default: none
+
+    - kernel-add-ppa
+      (boolean; default: True)
+      controls if the snappy-dev PPA should be added to the system
+
+    - kernel-use-llvm
+      (boolean or string to specify version suffix; default: False)
+      Use the LLVM substitutes for the GNU binutils utilities. Set this to a
+      string (e.g. "-12") to use a specific version of the LLVM utilities.
+
+This plugin support cross compilation, for which plugin expects
+the build-environment is comfigured accordingly and has foreign architectures
+setup accordingly.
+"""
+
+import logging
+import os
+import re
+import subprocess
+import sys
+from typing import Any, Dict, List, Optional, Set
+
+from overrides import overrides
+
+from snapcraft_legacy.internal.repo import errors
+from snapcraft_legacy.plugins.v2 import PluginV2
+from snapcraft_legacy.project._project_options import ProjectOptions
+
+logger = logging.getLogger(__name__)
+
+_compression_command = {"gz": "gzip", "lz4": "lz4", "xz": "xz", "zstd": "zstd"}
```

Comment:
care to define that common place?

Reply:
I didn't try defining it because I'm actually not sure on the "policy" regarding code that will be used by both `snapcraft` and `snapcraft_legacy`. I see that there is some code in `snapcraft` that imports from `snapcraft_legacy`, so maybe `snapcraft_legacy.kernel`?